build/yaml/templates/component-template.yml

```diff
@@ -0,0 +1,44 @@
+parameters:
+- name: componentType
+  displayName: 'What type of component?'
+  type: string
+  values:
+  - declarativeAsset
+  - codeExtension
+  - generator
+
+stages:
+  - ${{ if eq(parameters.componentType, 'codeExtension') }}: 
+    - stage: Build_pack
+      displayName: Build and package
+      jobs:
+        - job: Build
+          displayName: Build and package
+          steps:
+          - template: dotnet-package-steps.yml
+          - template: nuget-build-artifacts-steps.yml
```

Comment:
The PublishBuildArtifacts@1 task is in both the dotnet-package-steps and nuget-build-artifacts-steps files. Is that intentional?

Reply:
Yes so that they can share the same logic. This will likely be refactored when introducing the signing/staging steps.

Reply:
Pulled this into the dotnet/nuspec templates